### PR TITLE
Migrate py37 programs to py3

### PR DESCRIPTION
**Pull request by freezeboy**
###### Motivation for this change

Some programs in nixpkgs were using explicitly python37, and they also work with latest branch, so here are some "migrations" that seem correct.

I started the binaries they look ok

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -2490,7 +2490,7 @@ in
 
   psstop = callPackage ../tools/system/psstop { };
 
-  precice = callPackage ../development/libraries/precice { python3 = python37; };
+  precice = callPackage ../development/libraries/precice { python3 = python3; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:2493]: This is redundant.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -937,7 +937,7 @@ in
 
   iamy = callPackage ../tools/admin/iamy { };
 
-  azure-cli = callPackage ../tools/admin/azure-cli { python = python37; };
+  azure-cli = callPackage ../tools/admin/azure-cli { python = python3; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:940]: Using python3 in the package is probably more straight forward.

**Comment by SuperSandro2000** [comment]
When trying to build this with nixpkgs-review I get several thousands rebuilds. Not sure what went wrong here.

**Comment by freezeboy** [comment]
@SuperSandro2000 azure-cli has its own package set.. maybe redundant with other azure-* packages by the way

@jonringer Do you know why there is a special python-package set ?

**Comment by jonringer** [comment]
> @jonringer Do you know why there is a special python-package set ?

because the azure-cli lags 1-6 weeks behind most of the python azure sdk, or sometimes uses pre-releases for certain packages. azure-cli would essentially be unusable without it.

**Comment by jonringer** [comment]
azure-cli was kept at python37 originally because aiohttp didn't support it for the longest time

**Comment by freezeboy** [comment]
ok good to know

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103742` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>azure-cli</li>
    <li>html5validator</li>
    <li>manim</li>
    <li>precice</li>
  </ul>
</details>

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 103742 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>5 packages built:</summary>  <ul>    <li>azure-cli</li>    <li>html5validator</li>    <li>manim</li>    <li>precice</li>    <li>syncplay</li>  </ul></details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103742` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>azure-cli</li>
    <li>html5validator</li>
    <li>manim</li>
    <li>precice</li>
  </ul>
</details>

**Pull request merged**